packages/artifact/src/internal/download-http-client.ts

```diff
@@ -263,26 +311,47 @@ export class DownloadHttpClient {
       if (isGzip) {
         const gunzip = zlib.createGunzip()
         response.message
+          .on('error', error => {
+            core.error(
+              `An error occurred while attempting to read the response stream`
+            )
+            reject(error)
+            gunzip.close()
+          })
           .pipe(gunzip)
+          .on('error', error => {
+            core.error(
+              `An error occurred while attempting to decompress the response stream`
+            )
+            reject(error)
+            destinationStream.close()
```

Comment:
```suggestion
            destinationStream.close()
            reject(error)
```

We should close the stream first before erroring and retrying.